app/src/main/java/co/runloop/influencer/fragment/SmsAuthFragment.java

```diff
@@ -0,0 +1,172 @@
+                }
+            }
+        });
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater,
+                             @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        View root = inflater.inflate(R.layout.frag_sms_auth, container, false);
+
+        phoneNumberEt = root.findViewById(R.id.frag_sms_auth_phone_number_et);
+        TextWatcher watcher = new PhoneNumberFormattingTextWatcher(defLocale.getCountry()) {
+            @Override
+            public synchronized void afterTextChanged(Editable s) {
+                super.afterTextChanged(s);
+                updatePhoneNumberValidity(s.toString());
+            }
+        };
+        phoneNumberEt.addTextChangedListener(watcher);
+
+        submitPhoneNumberBtn = root.findViewById(R.id.frag_sms_auth_submit_phone_btn);
+        submitPhoneNumberBtn.setOnClickListener(view -> {
+            String number = PhoneNumberUtils.formatNumberToE164(
+                    phoneNumberEt.getText().toString(),
+                    defLocale.getCountry());
+            smsAuthViewModel.submitPhoneNumber(number);
+        });
+
+        codeEt = root.findViewById(R.id.frag_sms_auth_sms_code_et);
+        submitCodeBtn = root.findViewById(R.id.frag_sms_auth_submit_code_btn);
+        submitCodeBtn.setOnClickListener(view -> {
+            smsAuthViewModel.submitConfirmCode(codeEt.getText().toString().trim());
+        });
+
+        if (ContextCompat.checkSelfPermission(App.get().getApplicationContext(),
+                Manifest.permission.READ_PHONE_STATE) == PackageManager.PERMISSION_DENIED) {
+            requestPermissions(new String[]{Manifest.permission.READ_PHONE_STATE}, READ_PHONE_STATE_PERM_RC);
+        } else {
+            retrievePhoneNumber();
+        }
+        updatePhoneNumberValidity(phoneNumberEt.getText().toString());
+
+        return root;
+    }
+
+    private void updatePhoneNumberValidity(String phoneNumber) {
+        if (PhoneNumberUtils.isPossibleNumber(phoneNumber, defLocale.getCountry())) {
+            submitPhoneNumberBtn.setEnabled(true);
+            phoneNumberEt.setTextColor(Color.BLACK);
+        } else {
+            submitPhoneNumberBtn.setEnabled(false);
+            phoneNumberEt.setTextColor(Color.RED);
+        }
+    }
+
+    @SuppressWarnings({"MissingPermission", "HardwareIds"})
+    private void retrievePhoneNumber() {
+        TelephonyManager telephonyManager = (TelephonyManager)
+                App.get().getApplicationContext().getSystemService(Context.TELEPHONY_SERVICE);
+        phoneNumberEt.setText(telephonyManager.getLine1Number());
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode,
+                                           @NonNull String[] permissions,
+                                           @NonNull int[] grantResults) {
+        if (requestCode == READ_PHONE_STATE_PERM_RC) {
```

Comment:
This permission is optional, it needed only for autofill phone number, do you think we need to log it ?

Reply:
It's not critical either way. In the past, logging things have helped saved me some time on debug, but of course, too many logs can be counter-productive... 

Since you put some thought into it, I leave it as your decision.